cmd/broker/ingress/main.go

```diff
@@ -115,6 +126,8 @@ func main() {
 		logger.Error("manager.Start() returned an error", zap.Error(err))
 	}
 	logger.Info("Exiting...")
+	wg.Wait()
```

Comment:
I think this has to be time bounded. I don't want a single odd server to hold up the binary from restarting indefinitely.

Reply:
Kubernetes takes care of this for us. The default grace period is 30 seconds, after which a container is forcibly terminated. https://kubernetes.io/docs/concepts/workloads/pods/pod/#termination-of-pods

If we want native shutdown timeout, something like this might work:

```go
ctx := context.WithTimeout(context.Background(), time.Second*30)
go func() {
  select {
  case <-ctx.Done():
    log.Fatal("shutdown took too long")
  }
}()
```

But I think I prefer relying on Kubernetes in this case, since that gives the operator control over the timeouts without the code being involved.

Reply:
My preference is to do both. We can set the timeout here to something relatively high, but I like the guarantee of a bound.

In fact, if something in the manager returns an error from its `Start()`, then the manager will initiate the shutdown. This will happen without K8s being aware. So if something else in the manager doesn't shutdown, K8s won't help. Right?

Reply:
Good point, I'll add a shutdown timeout here.

My hope is that this will only be necessary until the Manager is fixed to have reasonable shutdown behavior.